### PR TITLE
APP-903  test AllowlistVotingSetup and ERC20VotingSetup

### PR DESCRIPTION
**Pull request by Rekard0**
## Description

- Test AllowlistVotingSetup.
- Test ERC20VotingSetup.

Task: [APP-903](https://aragonassociation.atlassian.net/browse/APP-903)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: packages/contracts/test/voting/erc20-voting-setup.ts [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+          Op.Grant,
+          anticipatedMerkleMinterAddress,
+          targetDao.address,
+          AddressZero,
+          UPGRADE_PERMISSION_ID,
+        ],
+      ]);
+    });
+
+    it('correctly setups the plugin and helpers, when a token address is not passed', async () => {
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/voting/erc20-voting-setup.ts:427]: "setups" is not a verb. 
"it sets up"

File: packages/contracts/test/voting/erc20-voting-setup.ts [hunk 2]
```diff
@@ -0,0 +1,641 @@
...
+
+      const permissions1 =
+        await erc20VotingSetup.callStatic.prepareUninstallation(
+          targetDao.address,
+          plugin,
+          [token.address],
+          EMPTY_DATA
+        );
+
+      const essencialPermissions = [
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/voting/erc20-voting-setup.ts:572]:
```suggestion
      const essentialPermissions = [
```

File: packages/contracts/test/voting/erc20-voting-setup.ts [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+          Op.Revoke,
+          targetDao.address,
+          plugin,
+          AddressZero,
+          EXECUTE_PERMISSION_ID,
+        ],
+      ];
+
+      expect(permissions1.length).to.be.equal(3);
+      expect(permissions1).to.deep.equal([...essencialPermissions]);
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/voting/erc20-voting-setup.ts:597]:
```suggestion
      expect(permissions1).to.deep.equal([...essentialPermissions]);
```

File: packages/contracts/test/voting/erc20-voting-setup.ts [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+        await erc20VotingSetup.callStatic.prepareUninstallation(
+          targetDao.address,
+          plugin,
+          [token.address, merkleMinter.address],
+          EMPTY_DATA
+        );
+
+      expect(permissions2.length).to.be.equal(7);
+      expect(permissions2).to.deep.equal([
+        ...essencialPermissions,
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/voting/erc20-voting-setup.ts:609]:
```suggestion
        ...essentialPermissions,
```

**Review by Michael-A-Heuer — DISMISSED** [review]

File: packages/contracts/test/voting/erc20-voting-setup.ts (hunk 1 above)
Comment (Rekard0) [reply packages/contracts/test/voting/erc20-voting-setup.ts:427]: Thanks,
Amended

File: packages/contracts/test/voting/erc20-voting-setup.ts (hunk 2 above)
Comment (Rekard0) [reply packages/contracts/test/voting/erc20-voting-setup.ts:572]: Amended

**Review by novaknole — APPROVED** [review]

**Review by Michael-A-Heuer — APPROVED** [review]

**Pull request merged**